### ISSUE TITLE
Investigate task queuing

### ISSUE THREAD

**Issue by hndrewaall** [massgo/league#61]
We will soon need the ability to schedule and queue jobs. Some examples of things we might want it to handle:
- Sending email
- Manage backups
- Perform statistical computations (update ratings etc.)
- Generate graphs

Some ideas for software:
- https://airflow.incubator.apache.org/ : generic DAG task manager. Possibly overkill. Can use Celery for workers.
- http://www.celeryproject.org/ : might be easiest way for executing Python code, not sure how easy it is to use for non-Python tasks (might require Airflow or similar)

**Comment by wisygig** [comment]
I'm willing to take a look at this.

**Comment by hndrewaall** [comment]
Cool! All thoughts welcome

**Comment by hwchen** [comment]
I'm also checking this out.

**Comment by hwchen** [comment]
Quick thoughts.

The benefit of something like airflow for us seems to be that it allows for
scheduling based on dependencies. But this comes with more overhead in
understanding the system, and I don't know how complicated our workflows
will be.

I don't know if something like a cron job would become difficult to
maintain, but might be simpler overall. (I have little experience using
cron, so please let me know if I'm wrong).

**Comment by hndrewaall** [comment]
Yeah, I think Airflow is definitely overkill. Cron is fine for scheduling, but it has two problems:

1. It doesn't "cluster" well. Ideally, if we do something cron-like, it'd at least be dockerizable. There are a few options that accomplish this

2. It doesn't allow us to trigger events from application code. I really want to both have scheduled tasks and event-driven tasks. Possibly we should be using two separate tools for these

**Comment by hndrewaall** [comment]
For me the appeal of Airflow was mainly that it could use Celery workers, and I know Celery is good for managing event-driven tasks

**Comment by hndrewaall** [comment]
Some cron-like functionality for Celery: http://docs.celeryproject.org/en/latest/userguide/periodic-tasks.html

**Comment by hndrewaall** [comment]
@hwchen do you think you'll be able to look at this in the next week or two?
